### ISSUE TITLE
[BUG] Crash when turning on debug logging

### ISSUE THREAD

**Issue by jgabaut** [jgabaut/koliseo#22]
**Describe the bug**
It seems that the variadic format of `kls_log()` doesn't go well with some Linux versions.
On any attempt to call the function, and subsequently `vfprintf()`, which is the actual culprit, a crash happens because of missing support.
If the platform really is the cause, this would effectively lock some distros from debugging for versions `>0.1.11`, since before `0.1.12` `kls_log()` was not variadic.

## Update

I removed mentions of `32bit Linux` since I don't believe that's the culprit for this one.

Useful backtrace line was:

```
 with __strlen_avx2 () at ../sysdeps/x86_64/multiarch/strlen-avx2.S:74 74	
  ../sysdeps/x86_64/multiarch/strlen-avx2.S: No such file or directory.
#1  0x00007ffff7c76db1 in __vfprintf_internal (_..._)
    at ./stdio-common/vfprintf-internal.c:1517
```

This [link](https://stackoverflow.com/questions/35104593/vfprintf-crash-on-32-bit-linux) mentions something similar but doesn't ultimately pin the problem to a platform.
[This comment](https://stackoverflow.com/questions/35104593/vfprintf-crash-on-32-bit-linux#comment57980984_35104593) seems to suggest it's about printing stuff.

The issue may be related to printing sized stuff which will have a different size on some platforms.

**To Reproduce**

1. Compile with debug on.
2. Run the demo.
3. Demo crashes as soon at the first `kls_log()` call.

**Expected behavior**
Log function prints to debug log file and returns peacefully.

 - OS: `some Linux`
 - Version: `>=0.1.12`

**Comment by jgabaut** [comment]
Updated the issue to remove mentions of 32bit Linux.

**Comment by jgabaut** [comment]
As per [this comment](https://github.com/jgabaut/helapordo/issues/20#issuecomment-1815636636), maybe the cause of this could lie in a specific `docker` image configuration.

**Comment by jgabaut** [comment]
Looking into this again, it seems this may have been fixed already in `0.2.4`.

Will do more blaming when I'm home and will report to the relevant issue in `helapordo`.

**Comment by jgabaut** [comment]
Seems this was fixed in `0.2.4`, but I still got to reference the fix itself (it was a bad call to `kls_log()`, with a format string expecting a missing integer argument). Closing for now, will add that later.
